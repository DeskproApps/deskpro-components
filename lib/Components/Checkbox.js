'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _InputLabel = require('./InputLabel');

var _InputLabel2 = _interopRequireDefault(_InputLabel);

var _newid = require('./utils/newid');

var _newid2 = _interopRequireDefault(_newid);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Checkbox = (_temp2 = _class = function (_React$Component) {
  _inherits(Checkbox, _React$Component);

  function Checkbox() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Checkbox.__proto__ || Object.getPrototypeOf(Checkbox)).call.apply(_ref, [this].concat(args))), _this), _this.handleChange = function (event) {
      _this.props.onChange(event.target.checked, _this.props.value);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Checkbox, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      if (this.props.id) {
        this.id = this.props.id;
      } else {
        this.id = (0, _newid2.default)();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          children = _props.children,
          className = _props.className,
          value = _props.value,
          elementProps = _objectWithoutProperties(_props, ['children', 'className', 'value']);

      var props = Object.assign({}, elementProps);
      delete props.onChange;
      return _react2.default.createElement(
        'div',
        {
          className: (0, _classnames2.default)('dp-input--checkbox', className, { 'dp-input--checkbox--disabled': this.props.disabled })
        },
        _react2.default.createElement('input', _extends({
          type: 'checkbox',
          value: value,
          id: this.id,
          ref: function ref(c) {
            _this2.input = c;
          },
          onChange: this.handleChange
        }, elementProps)),
        _react2.default.createElement('label', { htmlFor: this.id, className: 'dp-input--checkbox__checkbox' }),
        _react2.default.createElement(
          _InputLabel2.default,
          { htmlFor: this.id },
          children
        )
      );
    }
  }]);

  return Checkbox;
}(_react2.default.Component), _class.propTypes = {
  className: _propTypes2.default.string,
  disabled: _propTypes2.default.bool,
  id: _propTypes2.default.string,
  value: _propTypes2.default.oneOf(_propTypes2.default.string, _propTypes2.default.number),
  children: _propTypes2.default.node,
  onChange: _propTypes2.default.func
}, _class.defaultProps = {
  onChange: function onChange() {}
}, _temp2);
exports.default = Checkbox;