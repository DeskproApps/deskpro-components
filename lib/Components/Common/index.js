'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListElement = exports.List = undefined;

var _List = require('./List');

var _ListElement = require('./ListElement');

exports.List = _List.List;
exports.ListElement = _ListElement.ListElement;