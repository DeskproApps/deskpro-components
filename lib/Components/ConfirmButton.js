'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactOnclickoutside = require('react-onclickoutside');

var _reactOnclickoutside2 = _interopRequireDefault(_reactOnclickoutside);

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ClickOutsideButton = (0, _reactOnclickoutside2.default)(_Button2.default);
var ConfirmButton = (_temp = _class = function (_React$Component) {
  _inherits(ConfirmButton, _React$Component);

  function ConfirmButton(props) {
    _classCallCheck(this, ConfirmButton);

    var _this = _possibleConstructorReturn(this, (ConfirmButton.__proto__ || Object.getPrototypeOf(ConfirmButton)).call(this, props));

    _this.state = {
      confirm: false
    };
    _this.clickOutside = _this.clickOutside.bind(_this);
    _this.getLabel = _this.getLabel.bind(_this);
    _this.handleClick = _this.handleClick.bind(_this);
    return _this;
  }

  _createClass(ConfirmButton, [{
    key: 'getLabel',
    value: function getLabel() {
      if (this.state.confirm) {
        return this.props.message;
      }
      return this.props.children;
    }
  }, {
    key: 'handleClick',
    value: function handleClick(e) {
      e.preventDefault();
      if (!this.props.disabled) {
        if (this.state.confirm) {
          this.props.onClick(e);
          this.setState({
            confirm: false
          });
        } else {
          this.setState({
            confirm: true
          });
        }
      }
    }
  }, {
    key: 'clickOutside',
    value: function clickOutside() {
      this.setState({
        confirm: false
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var props = Object.assign({}, this.props);
      delete props.message;
      delete props.onClick;
      return _react2.default.createElement(
        ClickOutsideButton,
        _extends({
          onClick: this.handleClick,
          onClickOutside: this.clickOutside,
          disableOnClickOutside: !this.state.confirm
        }, props),
        this.getLabel()
      );
    }
  }]);

  return ConfirmButton;
}(_react2.default.Component), _class.propTypes = {
  children: _propTypes2.default.node,
  message: _propTypes2.default.string,
  disabled: _propTypes2.default.bool,
  onClick: _propTypes2.default.func
}, _class.defaultProps = {
  disabled: false,
  message: 'Are you sure?',
  onClick: function onClick() {}
}, _temp);
exports.default = ConfirmButton;