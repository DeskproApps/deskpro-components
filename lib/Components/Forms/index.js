'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = exports.LabelInput = exports.Label = exports.InputLabel = exports.Input = exports.Checkbox = undefined;

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

var _Input = require('./Input');

var _Input2 = _interopRequireDefault(_Input);

var _InputLabel = require('./InputLabel');

var _InputLabel2 = _interopRequireDefault(_InputLabel);

var _Label = require('./Label');

var _Label2 = _interopRequireDefault(_Label);

var _LabelInput = require('./LabelInput');

var _LabelInput2 = _interopRequireDefault(_LabelInput);

var _Select = require('./Select');

var _Select2 = _interopRequireDefault(_Select);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Checkbox = _Checkbox2.default;
exports.Input = _Input2.default;
exports.InputLabel = _InputLabel2.default;
exports.Label = _Label2.default;
exports.LabelInput = _LabelInput2.default;
exports.Select = _Select2.default;