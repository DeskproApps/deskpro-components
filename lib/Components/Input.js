'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;
// import SVGInline from 'react-svg-inline';


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _newid = require('./utils/newid');

var _newid2 = _interopRequireDefault(_newid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// import LoadingSvg from 'styles/images/input_loading.svg';
// import ValidatedSvg from 'styles/images/tick.svg';

var Input = (_temp = _class = function (_React$Component) {
  _inherits(Input, _React$Component);

  function Input(props) {
    _classCallCheck(this, Input);

    var _this = _possibleConstructorReturn(this, (Input.__proto__ || Object.getPrototypeOf(Input)).call(this, props));

    _this.onChange = function (event) {
      _this.props.onChange(event.currentTarget.value || '');
    };

    _this.onFocus = function (e) {
      _this.setState({
        focus: true
      });
      _this.props.onFocus(e);
    };

    _this.onBlur = function (e) {
      _this.setState({
        focus: false
      });
      _this.props.onBlur(e);
    };

    _this.getIcon = function () {
      var icon = _this.props.icon;

      if (!icon) {
        return null;
      }
      return _react2.default.createElement('i', { className: (0, _classnames2.default)('fa', 'fa-' + icon) });
    };

    _this.getPrefix = function () {
      var prefix = _this.props.prefix;

      if (!prefix) {
        return null;
      }
      return _react2.default.createElement(
        'div',
        { className: 'dp-input__prefix' },
        prefix
      );
    };

    _this.getSuffix = function () {
      var suffix = _this.props.suffix;

      if (!suffix) {
        return null;
      }
      return _react2.default.createElement(
        'div',
        { className: 'dp-input__suffix' },
        suffix
      );
    };

    _this.focus = function () {
      _this.input.focus();
    };

    _this.state = {
      focus: false
    };
    return _this;
  }

  _createClass(Input, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      if (this.props.id) {
        this.id = this.props.id;
      } else {
        this.id = (0, _newid2.default)();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          className = _props.className,
          validated = _props.validated,
          validating = _props.validating,
          icon = _props.icon,
          prefix = _props.prefix,
          suffix = _props.suffix,
          elementProps = _objectWithoutProperties(_props, ['className', 'validated', 'validating', 'icon', 'prefix', 'suffix']);

      var props = Object.assign({}, elementProps);
      delete props.onChange;
      delete props.onFocus;
      delete props.onBlur;
      return _react2.default.createElement(
        'div',
        {
          className: (0, _classnames2.default)('dp-input', className, {
            'dp-input--validating': validating,
            'dp-input--validated': validated,
            'dp-input--with-icon': icon,
            'dp-input--with-prefix': prefix,
            'dp-input--with-suffix': suffix,
            'dp-input--focused': this.state.focus
          })
        },
        this.getPrefix(),
        this.getIcon(),
        _react2.default.createElement('input', _extends({
          id: this.id,
          ref: function ref(c) {
            _this2.input = c;
          },
          onChange: this.onChange,
          onFocus: this.onFocus,
          onBlur: this.onBlur
        }, props)),
        this.getSuffix()
      );
    }
  }]);

  return Input;
}(_react2.default.Component), _class.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  label: _propTypes2.default.string,
  id: _propTypes2.default.string,
  icon: _propTypes2.default.string,
  prefix: _propTypes2.default.string,
  suffix: _propTypes2.default.string,
  validated: _propTypes2.default.bool,
  validating: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  readonly: _propTypes2.default.bool,
  required: _propTypes2.default.bool,
  onChange: _propTypes2.default.func,
  onFocus: _propTypes2.default.func,
  onBlur: _propTypes2.default.func
}, _class.defaultProps = {
  onChange: function onChange() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
}, _temp);
exports.default = Input;