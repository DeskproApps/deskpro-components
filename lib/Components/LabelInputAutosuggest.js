'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _class, _temp;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Label = require('./Label');

var _Label2 = _interopRequireDefault(_Label);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var LabelInputAutosuggest = (_temp = _class = function (_LabelInput) {
  _inherits(LabelInputAutosuggest, _LabelInput);

  function LabelInputAutosuggest() {
    _classCallCheck(this, LabelInputAutosuggest);

    return _possibleConstructorReturn(this, (LabelInputAutosuggest.__proto__ || Object.getPrototypeOf(LabelInputAutosuggest)).apply(this, arguments));
  }

  return LabelInputAutosuggest;
}(_Label2.default), _class.propTypes = {
  labels: _propTypes2.default.array,
  options: _propTypes2.default.array
}, _temp);
exports.default = LabelInputAutosuggest;