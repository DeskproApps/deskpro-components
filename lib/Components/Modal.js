'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Modal = (_temp = _class = function (_React$Component) {
  _inherits(Modal, _React$Component);

  function Modal(props) {
    _classCallCheck(this, Modal);

    var _this = _possibleConstructorReturn(this, (Modal.__proto__ || Object.getPrototypeOf(Modal)).call(this, props));

    _this.updateWindowDimensions = function () {
      _this.setState({
        width: window.innerWidth,
        height: window.innerHeight
      });
    };

    _this.state = {
      width: 0,
      height: 0
    };
    return _this;
  }

  _createClass(Modal, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      this.updateWindowDimensions();
      window.addEventListener('resize', function () {
        if (!_this2.ticking) {
          window.requestAnimationFrame(function () {
            _this2.updateWindowDimensions();
            _this2.ticking = false;
          });
        }
        _this2.ticking = true;
      });
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('resize', this.updateWindowDimensions);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          title = _props.title,
          children = _props.children,
          buttons = _props.buttons,
          style = _props.style,
          closeModal = _props.closeModal;
      var _state = this.state,
          width = _state.width,
          height = _state.height;

      var modalStyle = Object.assign({}, style);
      modalStyle.maxHeight = height * 0.7;
      modalStyle.maxWidth = width * 0.7;
      return _react2.default.createElement(
        'div',
        { className: 'dp-modal__overlay' },
        _react2.default.createElement(
          'div',
          {
            className: 'dp-modal',
            style: modalStyle
          },
          _react2.default.createElement(
            'div',
            { className: 'dp-modal__title' },
            title,
            _react2.default.createElement(
              'div',
              { className: 'dp-modal__close', onClick: closeModal },
              'X'
            )
          ),
          _react2.default.createElement(
            'div',
            { className: 'dp-modal__body' },
            children
          ),
          _react2.default.createElement(
            'div',
            { className: 'dp-modal__buttons' },
            buttons
          )
        )
      );
    }
  }]);

  return Modal;
}(_react2.default.Component), _class.propTypes = {
  title: _propTypes2.default.string,
  children: _propTypes2.default.node,
  buttons: _propTypes2.default.node,
  style: _propTypes2.default.object,
  closeModal: _propTypes2.default.func
}, _class.defaultProps = {
  closeModal: function closeModal() {}
}, _temp);
exports.default = Modal;