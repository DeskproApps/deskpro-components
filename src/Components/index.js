export { default as Alert } from './Alert';
export { default as Avatar } from './Avatar';
export { default as Icon } from './Icon';
export { default as Modal } from './Modal';
export { default as Pane } from './Pane';
export { default as Progress } from './Progress';
export { default as Stars } from './Stars';
export { default as Urgency } from './Urgency';
