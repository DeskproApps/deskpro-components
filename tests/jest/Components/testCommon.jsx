import React from 'react';
import renderer from 'react-test-renderer';
import { Icon } from 'Components/index';
import {
  Count,
  Heading,
  Highlighter,
  QueryableList,
  List,
  ListElement,
  SelectableList,
  ToggleableList,
  Popper,
  Portal,
  Scrollbar
} from 'Components/Common/index';

// https://github.com/facebook/react/issues/7740#issuecomment-247335106
const createNodeMock = () => ({
  addEventListener: () => {}
});

describe('>>> Buttons --- Common', () => {
  it('+++capturing Snapshot of Count', () => {
    const renderedValue = renderer.create(
      <div>
        <Count>3</Count><br />
        <Count>3333</Count><br />
        <Count separator=".">3333</Count>
      </div>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Heading', () => {
    const renderedValue = renderer.create(
      <Heading size={3} icon="globe" underline controls={<div><Icon name="gear" /><Icon name="refresh" /></div>}>
        Size 3
      </Heading>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Highlighter', () => {
    const renderedValue = renderer.create(
      <Highlighter highlight="feedback">
        Android feedback (Support)
      </Highlighter>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of List', () => {
    const renderedValue = renderer.create(
      <List className="test-class">
        <ListElement name="item1">
          Item1
        </ListElement>
        <ListElement name="item2">
          Item2
        </ListElement>
      </List>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of QueryableList', () => {
    const renderedValue = renderer.create(
      <QueryableList className="test-class" whereName="@all">
        <ListElement name="item1">
          Item1
        </ListElement>
        <ListElement name="item2">
          Item2
        </ListElement>
      </QueryableList>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of SelectableList', () => {
    const renderedValue = renderer.create(
      <SelectableList className="test-class">
        <ListElement name="item1">
          Item1
        </ListElement>
        <ListElement name="item2">
          Item2
        </ListElement>
      </SelectableList>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of ToggleableList', () => {
    const renderedValue = renderer.create(
      <ToggleableList className="test-class" on="click" toggle="selected">
        <ListElement name="item1">
          Item1
        </ListElement>
        <ListElement name="item2">
          Item2
        </ListElement>
      </ToggleableList>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Popper', () => {
    const renderedValue = renderer.create(
      <Popper target="target" placement="bottom" opened>
        <p>Test</p>
      </Popper>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Scrollbar', () => {
    const renderedValue = renderer.create(
      <Scrollbar>
        <p>Test</p>
      </Scrollbar>,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Portal', () => {
    const renderedValue = renderer.create(
      <Portal>
        <p>Test</p>
      </Portal>,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
