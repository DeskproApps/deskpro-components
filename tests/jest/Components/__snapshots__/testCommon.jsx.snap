// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Buttons --- Common +++capturing Snapshot of Count 1`] = `
<div>
  <span
    className="dp-count"
  >
    3
  </span>
  <br />
  <span
    className="dp-count"
  >
    3,333
  </span>
  <br />
  <span
    className="dp-count"
  >
    3.333
  </span>
</div>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of Heading 1`] = `
<h3
  className="dp-heading dp-heading--underlined"
>
  <i
    className="fa fa-globe dp-icon dp-icon--s"
  />
  Size 3
  <span
    className="dp-heading__controls"
  >
    <div>
      <i
        className="fa fa-gear dp-icon dp-icon--s"
      />
      <i
        className="fa fa-refresh dp-icon dp-icon--s"
      />
    </div>
  </span>
</h3>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of Highlighter 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Android <i class=\\"dp-highlighted\\">feedback</i> (Support)",
    }
  }
/>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of List 1`] = `
<ul
  className="dp-list test-class"
>
  <li
    className="dp-list__element"
    name="item1"
  >
    Item1
  </li>
  <li
    className="dp-list__element"
    name="item2"
  >
    Item2
  </li>
</ul>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of Popper 1`] = `<div />`;

exports[`>>> Buttons --- Common +++capturing Snapshot of Portal 1`] = `<div />`;

exports[`>>> Buttons --- Common +++capturing Snapshot of QueryableList 1`] = `
<ul
  className="dp-list dp-queryable-list test-class"
>
  <li
    className="dp-list__element"
    name="item1"
  >
    Item1
  </li>
  <li
    className="dp-list__element"
    name="item2"
  >
    Item2
  </li>
</ul>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of Scrollbar 1`] = `
<div
  className="dp-scrollbar"
  style={
    Object {
      "height": "auto",
      "maxHeight": 200,
      "minHeight": 0,
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "bottom": undefined,
        "left": undefined,
        "marginBottom": 0,
        "marginRight": 0,
        "maxHeight": 200,
        "minHeight": 0,
        "overflow": "scroll",
        "position": "relative",
        "right": undefined,
        "top": undefined,
      }
    }
  >
    <p>
      Test
    </p>
  </div>
  <div
    className="dp-scrollbar__track-horizontal"
    style={
      Object {
        "display": "none",
        "height": 6,
        "position": "absolute",
      }
    }
  >
    <div
      className="dp-scrollbar__thumb-horizontal"
      style={
        Object {
          "display": "block",
          "height": "100%",
          "position": "relative",
        }
      }
    />
  </div>
  <div
    className="dp-scrollbar__track-vertical"
    style={
      Object {
        "display": "none",
        "position": "absolute",
        "width": 6,
      }
    }
  >
    <div
      className="dp-scrollbar__thumb-vertical"
      style={
        Object {
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of SelectableList 1`] = `
<ul
  className="dp-list dp-selectable-list test-class"
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOver={[Function]}
  tabIndex="-1"
>
  <li
    className="dp-list__element dp-selectable-list--selected"
    data-dp-index={0}
    name="item1"
    onClick={[Function]}
  >
    Item1
  </li>
  <li
    className="dp-list__element"
    data-dp-index={1}
    name="item2"
    onClick={[Function]}
  >
    Item2
  </li>
</ul>
`;

exports[`>>> Buttons --- Common +++capturing Snapshot of ToggleableList 1`] = `
<ul
  className="dp-list test-class"
>
  <li
    className="dp-list__element"
    data-dp-toggle-id={0}
    name="item1"
    onClick={[Function]}
    selected={false}
  >
    Item1
  </li>
  <li
    className="dp-list__element"
    data-dp-toggle-id={1}
    name="item2"
    onClick={[Function]}
    selected={false}
  >
    Item2
  </li>
</ul>
`;
